src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectManagementService.cs

```diff
@@ -0,0 +1,105 @@
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.ObjectManagement.Contracts;
+using Microsoft.SqlTools.Utility;
+
+namespace Microsoft.SqlTools.ServiceLayer.ObjectManagement
+{
+    /// <summary>
+    /// Main class for ObjectManagement Service functionality
+    /// </summary>
+    public class ObjectManagementService
+    {
+        private const string ObjectManagementServiceApplicationName = "azdata-object-management";
+        private static Lazy<ObjectManagementService> objectManagementServiceInstance = new Lazy<ObjectManagementService>(() => new ObjectManagementService());
+        public static ObjectManagementService Instance => objectManagementServiceInstance.Value;
+
+        public static ConnectionService connectionService;
+
+        private IProtocolEndpoint serviceHost;
+
+        public ObjectManagementService() { }
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                connectionService ??= ConnectionService.Instance;
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+            this.serviceHost = serviceHost;
+            this.serviceHost.SetRequestHandler(RenameRequest.Type, HandleRenameRequest);
+        }
+
+        /// <summary>
+        /// Method to handle the renaming operation
+        /// </summary>
+        /// <param name="requestParams">parameters which are needed to execute renaming operation</param>
+        /// <param name="requestContext">Request Context</param>
+        /// <returns></returns>
+        internal async Task HandleRenameRequest(RenameRequestParams requestParams, RequestContext<bool> requestContext)
+        {
+
+            Logger.Verbose("Handle Request in HandleProcessRenameEditRequest()");
+            ConnectionInfo connInfo;
+
+            if (connectionService.TryFindConnection(
+                    requestParams.ConnectionUri,
+                    out connInfo))
+            {
+                using (SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, ObjectManagementServiceApplicationName))
+                {
+
+                    IRenamable renameObject = this.GetSQLRenameObject(requestParams, sqlConn);
+
+                    renameObject.Rename(requestParams.NewName);
+                }
+            }
+            else
+            {
+                Logger.Error($"The connection {requestParams.ConnectionUri} could not be found.");
+                throw new Exception(SR.ErrorConnectionNotFound);
+            }
+
+            await requestContext.SendResult(true);
```

Comment:
does it must have a value? this is never going to be false.

Reply:
Yes, limitation of our request context stuff currently. I want to go fix that at some point to allow an empty (void) response but for now this is fine.